service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/C21OrderController.java

```diff
@@ -0,0 +1,122 @@
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.swagger.annotations.Api;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.document.domain.Document;
+import uk.gov.hmcts.reform.fpl.interfaces.C21CaseOrderGroup;
+import uk.gov.hmcts.reform.fpl.model.C21Order;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.common.DocmosisDocument;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentReference;
+import uk.gov.hmcts.reform.fpl.service.CreateC21OrderService;
+import uk.gov.hmcts.reform.fpl.service.DocmosisDocumentGeneratorService;
+import uk.gov.hmcts.reform.fpl.service.UploadDocumentService;
+import uk.gov.hmcts.reform.fpl.service.ValidateGroupService;
+
+import java.util.Map;
+
+import static org.apache.commons.lang3.StringUtils.defaultIfBlank;
+import static uk.gov.hmcts.reform.fpl.enums.DocmosisTemplates.C21;
+
+@Api
+@RequestMapping("/callback/create-order")
+@RestController
+public class C21OrderController {
+
+    private final ObjectMapper mapper;
+    private final DocmosisDocumentGeneratorService docmosisService;
+    private final UploadDocumentService uploadDocumentService;
+    private final CreateC21OrderService createC21OrderService;
+    private final ValidateGroupService validateGroupService;
+
+    @Autowired
+    public C21OrderController(ObjectMapper mapper,
+                              DocmosisDocumentGeneratorService docmosisService,
+                              UploadDocumentService uploadDocumentService,
+                              CreateC21OrderService createC21OrderService,
+                              ValidateGroupService validateGroupService) {
+        this.mapper = mapper;
+        this.docmosisService = docmosisService;
+        this.uploadDocumentService = uploadDocumentService;
+        this.createC21OrderService = createC21OrderService;
+        this.validateGroupService = validateGroupService;
+    }
+
+    @PostMapping("/about-to-start")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetails.getData())
+            .errors(validateGroupService.validateGroup(caseData, C21CaseOrderGroup.class))
```

Comment:
We need to discuss the above aproach, because CCD didn't really like this.

Reply:
@akomisarek Is this because CCD has changed something? This affects notice of proceedings also ://

Reply:
Yes CCD changed something which means we don't get a meaningful error message anymore for this kind of validation - left in for now as template expects a familyman number, can refactor to fix this + C6 later in another story if CCD don't resolve anything on their end